Apply PEP-563 (Postponed Evaluation of Annotations) to non-core airflow

potiuk opened the pull request:
This PR applies PEP-563 to all non-core airflow Python files - i.e.
those that we usually do not cherry-pick to v2-* branches. There
will be a follow-up to apply the "core changes" right after with
the intention to be cherry-picked to v2-* branch before 2.4.0
release candidate is ready, in order to make cherry-picking
easier. The follow up PR is #26290.

This PR is a result of combining some of the 47 PRs reviewed
and approved separately (otherwise it would have been unreviewable)

The history of those PRs can be changed in:
https://github.com/apache/airflow/pulls?q=is%3Apr+label%3Afuture-annotations+is%3Aopen

Relevant discussion: https://lists.apache.org/thread/81fr042s5d3v17v83bpo24tnrr2pp0fp
Lazy consensus call: https://lists.apache.org/thread/l74nvjh8tgbtojllhwkcn7f8mfnlz4jq


<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

uranusjr commented on airflow/__init__.py:
```diff
@@ -15,7 +15,7 @@
 # under the License.
-#
+from __future__ import annotations
 
```
Imports should _follow_ the docstring.

potiuk replied:
Fixed. I am also reviewing all other PRs to apply it too.

potiuk replied:
FYI -> I reviewed and updated all other "future annotations" PRs (actually some of them failed becaused of this problem).

potiuk committed:
Apply PEP-563 (Postponed Evaluation of Annotations) to non-core airflow

This PR applies PEP-563 to all non-core airflow Python files - i.e.
those that we usually do not cherry-pick to v2-* branches. There
will be a follow-up to apply the "core changes" right after with
the intention to be cherry-picked to v2-* branch before 2.4.0
release candidate is ready, in order to make cherry-picking
easier.

This PR is a result of combining some of the 47 PRs reviewed
and approved separately (otherwise it would have been unreviewable)

The history of those PRs can be changed in:
https://github.com/apache/airflow/pulls?q=is%3Apr+label%3Afuture-annotations+is%3Aopen

Relevant discussion: https://lists.apache.org/thread/81fr042s5d3v17v83bpo24tnrr2pp0fp
Lazy consensus call: https://lists.apache.org/thread/l74nvjh8tgbtojllhwkcn7f8mfnlz4jq
